packages/framework/src/Model/Product/Product.php

```diff
@@ -893,6 +905,109 @@ protected function setTranslations(ProductData $productData)
         }
     }
 
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductData $productData
+     */
+    private function setDomains($productData)
+    {
+        $domainIds = array_keys($productData->seoTitles);
+        foreach ($domainIds as $domainId) {
+            try {
+                $productDomain = $this->getProductDomain($domainId);
+            } catch (ProductDomainNotFoundException $e) {
+                $productDomain = new ProductDomain($this, $domainId);
+                $this->domains[$domainId] = $productDomain;
+            }
+            $productDomain->setSeoTitle($productData->seoTitles[$domainId]);
+            $productDomain->setSeoH1($productData->seoH1s[$domainId]);
+            $productDomain->setSeoMetaDescription($productData->seoMetaDescriptions[$domainId]);
+            $productDomain->setDescription($productData->descriptions[$domainId]);
+            $productDomain->setShortDescription($productData->shortDescriptions[$domainId]);
+        }
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Product\ProductDomain[]|\Doctrine\Common\Collections\ArrayCollection
+     */
+    public function getDomains()
+    {
+        return $this->domains;
+    }
+
+    /**
+     * @param int $domainId
+     * @return \Shopsys\FrameworkBundle\Model\Product\ProductDomain
+     */
+    public function getProductDomain(int $domainId)
```

Comment:
In Brand, method such as this wasn't public. The idea was that BrandDomain should look like an implementation detail from the outside.